### PR TITLE
Add timer docs

### LINKED ISSUE #27: Create timer documentation
Just a remember:
Do the timer() documentation

### PR DESCRIPTION
**Pull request by luanmuniz**
Fix #27
Can you take a look on this @leobalter ?

### CODE REVIEW

File: docs/methods/timer.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Dexter.timer()
+
+This function changes the default behaviour of setTimeout, setInterval, clearTimeout and clearInterval functions so you can manipulate the delay time through code.
```
Comment (leobalter) [inline docs/methods/timer.md]: `manipulate the delay time through code` => `manipulate time through code`

It should be good to add an extra paragraph mentioning the intended purpose. Like, that's good to make tests faster, but use it with caution.

File: docs/methods/timer.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Lets suppose that you set the setTimeout for 3000ms, with Dexter.timer you can use Timer.tick function to forcefully make the 3000ms advance instantly , or if you have a setInterval of 1000ms, you can forcefully make the function run as many times you want using Timer.tick()
```
Comment (leobalter) [inline docs/methods/timer.md]: `advance instantly , or` => `advance instantly, or` (space before ,)

File: docs/methods/timer.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+### Timer.tick(number)
+
+Tick is the function that forces time advance, it receives the number of miliseconds that you want to instantly advance. For example, if you have an instance of `setTimeout` or a `setInterval` with the timer set for 1000ms, you can call `Timer.tick(1000)` to make the instance run instantly.
+
+### Timer.restore()
+
+This function restores setTimeout, setInterval, clearTimeout and clearInterval to their original behaviours.
```
Comment (leobalter) [inline docs/methods/timer.md]: also: mention that `.restore` will also clear all the fake setTimeouts and setIntervals, not moving them to the real world.

File: docs/methods/timer.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Dexter.timer()
+
+This function changes the default behaviour of setTimeout, setInterval, clearTimeout and clearInterval functions so you can manipulate the delay time through code.
+
+Lets suppose that you set the setTimeout for 3000ms, with Dexter.timer you can use Timer.tick function to forcefully make the 3000ms advance instantly , or if you have a setInterval of 1000ms, you can forcefully make the function run as many times you want using Timer.tick()
+
```
Comment (leobalter) [inline docs/methods/timer.md:6]: and as mentioned on the `.restore`, it's good to mention the timer won't affect previously set timeouts and intervals.

**Comment by luanmuniz** [comment]
@leobalter can you write the extra paragraph mentioning the intended purpose? 
You propose the feature, i just code it. You know better than me

**Comment by leobalter** [comment]
Np, thank you

**Pull request merged**